Changes and updates in L1 & L2 (#1)
pupilfirst/ev101-scripts: L3 Mathematical modelling of an electric vehicle /Calculating Energy Consumption {Software simulation in Scilab and Xcos} - Part 1.md

The parameters of the WLTC Class 3 cycle are summarized in the table below:

|                                   | Low   | Medium | High  | Extra High | Total |
|-----------------------------------|-------|--------|-------|------------|-------|
| Duration, s                       | 589   | 433    | 455   | 323        | 1800  |
| Stop duration, s                  | 150   | 49     | 31    | 8          | 235   |
| Distance, m                       | 3095  | 4756   | 7162  | 8254       | 23266 |
| % of stops                        | 26.5% | 11.1%  | 6.8%  | 2.2%       | 13.4% |
| Maximum speed, km/h               | 56.5  | 76.6   | 97.4  | 131.3      |       |
| Average speed without stops, km/h | 25.3  | 44.5   | 60.7  | 94.0       | 53.5  |
| Average speed with stops, km/h    | 18.9  | 39.4   | 56.5  | 91.7       | 46.5  |
| Minimum acceleration, m/s^2^        | -1.5  | -1.5   | -1.5  | -1.44      |       |
| Maximum acceleration, m/s^2^        | 1.611 | 1.611  | 1.666 | 1.055      |       |

The method to calculate the energy consumption is straightforward, and it makes use of the Scilab/Xcos simulation environment. 

----

> >Everything explained in the video above is explained below as well, so that it is easier to follow and understand. if you are confused, please watch the video again. 

Reviewer: ```suggestion
>Everything explained in the video above is explained below as well, so that it is easier to follow and understand. If you are confused about anything that is covered in the video, we encourage you to watch the video again. 
```

## 1. Setting up the Simulation Environment

> We will now need to install and Setup Scilab, Follow the steps below to install and set it up.

Reviewer: ```suggestion
We will now need to install and setup Scilab. Follow the steps below to install and set it up.
```

### Step 1. Download Scilab

Click [here](https://www.scilab.org/download/6.1.0) to go to Scilab website, under the *Download* section and choose your installation package. For windows operating systems the are two versions: 32 and 64-bit. Download the one matching your operating system specification.

When the download is complete you’ll have on your disk drive a *.exe file (e.g. Scilab-5.4.0.exe)

> ### Step 2: Run *.exe file

Reviewer: These need not be headings since many of these do not have any explanation under it.

### Step 3: Select the language to be used during the installation process

### Step 4: Launch the Scilab Setup Wizard

### Step 5: Read the License Agreement

### Step 6: Review all installation settings

### Step 7: Launch Scilab
At the end of the installation, if you check the box for “Launch Scilab”, when you click the “Finish” button, Scilab will be launched.

Click the “Finish” button to exit the Setup Wizard.

![Scilab screen.jpg](https://www.pupilfirst.school/markdown_attachments/6316/wr3zglnbr5FLyIATVoNocQ)

When you launch Scilab you’ll get the start screen as in the image above

## 2. Building an Environment

Create a folder Named **"Electric vehicles"** on the desktop, or where-ever you like.

This will be the folder where all our project files are stored.

**Important note! unless all the required files are stored in the same folder, the simulation wont work.**

## 3. Acquiring the WLTC Test cycle Data

> Click on the link below to Download the data, Make sure you store it in  the *Electric vehicles* folder created in the previous step.

Reviewer: ```suggestion
Click on the link below to download the data. Make sure you store it in  the *Electric vehicles* folder created in the previous step.
```

>Important note: Only excel file version (2003) are handled.

### Let’s import the data describing the WLTP speed profile. The file can be found here[]:

The data will be first imported into Scilab, stored into a variable, and used later in Xcos for simulation purposes.

**Step 1**. Save the *.xls file after downloading, in the current Scilab working folder, i.e "Electric vehicles"

**Step 2**. Open the *.xls file and examine the data you want to import. 

![xls.jpg](https://demo.pflms.com/markdown_attachments/1781/qbn5dUHX0CuM6eMuLneloA)

In this example, the data is located in the second sheet, named **WLTC_class_3**. We will import the WLTP speed profile, which consists of time values and speed values. 

- **The time values**, in s, begin in the 8th row and 3rd column (C), with value 0. 

- **The speed values**, in kph, begin in the 8th row and 5th column (E), with value 0.0. If you scroll down towards the end of the table, you’ll see that the last data points are in the row 1808.

**Step 3**. Open SciNotes (Applications>Scinotes) and create a script file with the following content:

```
clear()
clc()

> //Decode ole file, extract and open Excel stream

Reviewer: This is the first mention of `ole` file in this lesson. We need to add a line about this or add an external link to Scilab website where this is discussed.
